azure/services/scalesets/scalesets.go

```diff
@@ -295,11 +309,19 @@ func (s *Service) patchVMSSIfNeeded(ctx context.Context, infraVMSS *azure.VMSS)
 		patch.Sku.Capacity = pointer.Int64(surge)
 	}
 
+	// If the VMSS is managed by an external autoscaler, we should patch the VMSS if customData has changed.
 	// If there are no model changes and no increase in the replica count, do not update the VMSS.
 	// Decreases in replica count is handled by deleting AzureMachinePoolMachine instances in the MachinePoolScope
-	if *patch.Sku.Capacity <= infraVMSS.Capacity && !hasModelChanges {
-		log.V(4).Info("nothing to update on vmss", "scale set", spec.Name, "newReplicas", *patch.Sku.Capacity, "oldReplicas", infraVMSS.Capacity, "hasChanges", hasModelChanges)
-		return nil, nil
+	if replicaseManagedByAutoscaler {
```

Comment:
this could be 

```
if *patch.Sku.Capacity <= infraVMSS.Capacity && !hasModelChanges && (!hasCustomDataChanges || !replicaseManagedByAutoscaler)  {
```

to avoid the duplication

Reply:
Consolidated if block